server/functions.lua

```diff
@@ -107,6 +107,75 @@ function QBCore.Functions.GetDutyCount(job)
     return count
 end
 
+--- Routingbucket stuff (Only touch if you know what you are doing)
+_G.QBCore.Player_Buckets = {} -- Bucket array containing all players that have been set to a different bucket
+_G.QBCore.Entity_Buckets = {} -- Bucket array containing all entities that have been set to a different bucket
```

Comment:
Whats the reasoning for adding these to the shared object? For this data to be up to date across all consuming resources, a resource would need to refresh the object before operations. It may be better to expose this data through some getter function that would copy the entity table state at runtime.

Also these data structures may be more efficient as a Map using entity ID as the key.

Reply:
Thanks for your taking your time to write the feedback btw, also from looking at the QBcore object, it made the most sense to add them there. Regarding the datastructure i was not so sure how to do it in a good way, so feel free to edit it and add whatever u believe is best